include/examples/simple_behavior.h

```diff
@@ -0,0 +1,192 @@
+   *          behavior is passed as an argument, permitting different
+   *          functionalities for different prior behaviors. In this example,
+   *          the behavior is marked as active and the initial robot state is
+   *          read.
+   * @param prev_behavior the previously active behavior
+   */
+  void Begin(std::unique_ptr<kodlab::Behavior<SimpleRobot>> &prev_behavior) override
+  {
+    // Update internal state
+    state_ = RUNNING;
+
+    // Read starting joint positions
+    joint_start_positions_ = robot_->GetJointPositions();
+    std::fprintf(stdout, "%s has begun\n", get_name().c_str());
+  }
+
+  /**
+   * @brief Updates the behavior
+   * @details This code runs after Begin, and is called on every control loop
+   *          update. Use this to define the primary content of your behavior,
+   *          e.g., to update and command new joint torques.  Additionally, this
+   *          code can monitor if the behavior has terminated and update flags
+   *          appropriately. In this example, this function reads the updated
+   *          joint velocities and maintains state machine corresponding to the
+   *          member enum `state_`.
+   */
+  void Update() override
+  {
+    // Update joint velocities from robot
+    joint_velocities_ = robot_->GetJointVelocities();
+
+    switch (state_)
+    {
+      case RUNNING:  // in a running state
+        // Set joint torques to running levels
+        desired_torques_ =
+            std::vector<float>(robot_->joints.size(), RUN_TORQUE);
+        break;
+
+      case STOPPING:  // in a stopping state
+        // Lower the joint torques to slowing levels
+        desired_torques_ =
+            std::vector<float>(robot_->joints.size(), SLOW_TORQUE);
+
+        // Set behavior to inactive if average velocity is below a threshold
+        float avg_velocity =
+            std::accumulate(joint_velocities_.begin(),
+                            joint_velocities_.end(),
+                            0.0) / static_cast<float>(joint_velocities_.size());
+        if (avg_velocity <= 0.1)
+        {
+          set_inactive();
+          std::fprintf(stdout, "%s is no longer active\n", get_name().c_str());
+        }
+        break;
+    }
+
+    // Set joint torques
+    robot_->SetTorques(desired_torques_);
+  }
+
+  /**
+   * @brief Stops the behavior
+   * @details This code is run when a behavior is directed to stop running by
+   *          the behavior manager. This should trigger a shutdown process for
+   *          the behavior, e.g. slowing the motor speed. The upcoming behavior
+   *          is passed as an argument to permit different shutdown procedures
+   *          for different on-deck behaviors. In this example, the internal
+   *          state is modified to indicate the behavior should stop.
+   * @param next_behavior the upcoming behavior
+   */
+  void Stop(std::unique_ptr<kodlab::Behavior<SimpleRobot>> &next_behavior) override
```

Comment:
Will stop be called once or multiple times?

Reply:
Stop is only called once.  The behavior switch sequence calls `Stop` on the running behavior, then  continues calling `Update` I'm that behavior until it's `ReadyToSwitch` function returns `true`, then switches to the new behavior and calls its `Begin` method.